### PR TITLE
[10.0] [MIG] product_customer_code_sale

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Product Customer Code For sale
==============================
Based on product_supplierinfo_for_customer, this module loads in every sale order the
customer code defined in the product.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

Commits are already preserved and can you please check the [history](https://github.com/nikul-serpentcs/sale-workflow/commits/10.0-product_customer_code_sale/product_customer_code_sale)

**Review by AaronHForgeFlow — COMMENTED** [review]
Please add the product-attribute repo in the oca_dependencies.txt file.

I'd also change the title of this PR, this is a migration not adding a new module.

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

File: product_customer_code_sale/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.multi
+    def _get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self:
```
Comment (AaronHForgeFlow) [inline product_customer_code_sale/models/sale.py]: You can improve performance by filtering here:

`for line in self.filtered(lambda sol: sol.product_id.supplier_ids):`

**Review by AaronHForgeFlow — COMMENTED** [review]
Tested on local and works properly.

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

File: product_customer_code_sale/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.multi
+    def _get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self.filtered(lambda sol: sol.product_id.supplier_ids):
+            product = line.product_id
+            if product.supplier_ids:
```
Comment (AaronHForgeFlow) [inline product_customer_code_sale/models/sale.py]: I think this check is not needed as you're filtering the lines

File: product_customer_code_sale/README.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============================
+Product Customer Code For sale
+==============================
+Based on product_supplierinfo_for_customer, this module loads in every sale order the
+customer code defined in the product.
+
```
Comment (AaronHForgeFlow) [inline product_customer_code_sale/README.rst:10]: I think the Usage section with the "Try me on runbot" button is a must:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

**Review by AaronHForgeFlow — COMMENTED** [review]
I tested on local and it looks good. Just come minor comment.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: It works. LGTM now

**Review by LoisRForgeFlow — APPROVED** [review]
Code LGTM. Could you please squash the commits of the migration into one before merge?

File: product_customer_code_sale/models/sale.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.multi
+    def _compute_get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self.filtered(lambda sol: sol.product_id.supplier_ids):
+            product = line.product_id
+            code_id = product_supplierinfo_obj.search([
```
Comment (AaronHForgeFlow) [inline product_customer_code_sale/models/sale.py:17]: This search is bad. The customer and supplierinfo of type 'customer' should be included as criteria.

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]

**Comment by nikul-serpentcs** [comment]
@aheficent Improved search method, Could you please review.

File: product_customer_code_sale/models/sale.py [hunk 5]
```diff
@@ -16,6 +16,7 @@ def _compute_get_product_customer_code(self):
             product = line.product_id
             code_id = product_supplierinfo_obj.search([
                 ('product_tmpl_id', '=', product.product_tmpl_id.id),
+                ('type', '=', 'customer'),
```
Comment (AaronHForgeFlow) [inline product_customer_code_sale/models/sale.py]: This is good. However I miss also the filter for the customer. `('name', '=', line.order_id.partner_id),` would do the thing I think

**Comment by nikul-serpentcs** [comment]
@aheficent Added filter for a customer, Could you please review.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Now it works as expected. Thanks

**Comment by rafaelbn** [comment]
Why this PR has been closed and not merged?

**Comment by AaronHForgeFlow** [comment]
@rafaelbn because the module had to be renamed (product customer code is deprecated in favor of product_supplierinfo_for_customer) and the author of this PR will not continue with this.

**Comment by rafaelbn** [comment]
OK thanks @aheficent !

**Pull request closed without merging**